### PR TITLE
switch to standalone communication

### PR DESCRIPTION
**Pull request by andreasmuelder**

### CODE REVIEW

**Comment by cdietrich** [comment]
how can i build the extension locally?

**Comment by andreasmuelder** [comment]
solidity-ide/releng/com.yakindu.solidity/releng> mvn clean verify

solidity-ide/extensions/vscode> gradlew vscodeExtension

**Comment by cdietrich** [comment]
hmm for me it works nicely. do you get any error messages?

**Comment by cdietrich** [comment]
and the java process is started as expected

```
MacDietrich:vscode[vscode_standalone] dietrich$ ps -ef | grep sol
  501   122     1   0 20Okt18 ??         0:45.85 /System/Library/CoreServices/loginwindow.app/Contents/MacOS/loginwindow console
  501 52264 52244   0 10:15pm ??         0:05.89 /Library/Java/JavaVirtualMachines/jdk1.8.0_181.jdk/Contents/Home/bin/java -classpath /Users/dietrich/.vscode/extensions/itemis.yakindu-solidity-ls-0.7.3/src/ls/etc:/Users/dietrich/.vscode/extensions/itemis.yakindu-solidity-ls-0.7.3/src/ls/repo/* -Dapp.name=solidity-ls -Dapp.pid=52264 -Dapp.repo=/Users/dietrich/.vscode/extensions/itemis.yakindu-solidity-ls-0.7.3/src/ls/repo -Dapp.home=/Users/dietrich/.vscode/extensions/itemis.yakindu-solidity-ls-0.7.3/src/ls -Dbasedir=/Users/dietrich/.vscode/extensions/itemis.yakindu-solidity-ls-0.7.3/src/ls org.eclipse.xtext.ide.server.ServerLauncher
  501 52505 40516   0 10:17pm ttys006    0:00.00 grep sol
```

**Comment by cdietrich** [comment]
bildschirmfoto 2018-10-27 um 22 19 07

**Comment by cdietrich** [comment]
you should first test the launcher manually

**Comment by andreasmuelder** [comment]
launcher is working, it is even working when starting the extension by F5. But when I install the extension I get the 
[Error - 10:28:00 PM] Connection to server is erroring. Shutting down server.

**Comment by cdietrich** [comment]
and the lauch from ~/.vscode/extensions/itemis.yakindu-solidity-ls-0.7.3/ is working?

**Comment by andreasmuelder** [comment]
Thats it! Didn't get is that there is a .vscode in the home directory after deleting it everything works :)

File: extensions/vscode/src/extension.ts [hunk 1]
```diff
@@ -1,41 +1,49 @@
...
+        documentSelector: ['Solidity'],
+        synchronize: {
+            fileEvents: workspace.createFileSystemWatcher('**/*.*')
+        }
+    };
+    
+    // Create the language client and start the client.
+    let lc = new LanguageClient('Solidity Language Server', serverOptions, clientOptions);
+    
+    var disposable2 =commands.registerCommand("sol.a.proxy", async () => {
```
Comment (cdietrich) [inline extensions/vscode/src/extension.ts]: these commands make no sense. we should remove this down to context.subscriptions.push(disposable2);

File: extensions/vscode/src/extension.ts [hunk 2]
```diff
@@ -1,41 +1,49 @@
...
+import { LanguageClient, LanguageClientOptions, ServerOptions } from 'vscode-languageclient';
+
+export function activate(context: ExtensionContext) {
+    // The server is a locally installed in src/mydsl
+    let launcher = os.platform() === 'win32' ? 'solidity-ls.bat' : 'solidity-ls';
+    let script = context.asAbsolutePath(path.join('src', 'ls', 'bin', launcher));
+    
+    let serverOptions: ServerOptions = {
+        run : { command: script },
+        debug: { command: script, args: ['-Xdebug','-Xrunjdwp:server=y,transport=dt_socket,address=8000,suspend=n,quiet=y','-Xmx256m'] }
```
Comment (cdietrich) [inline extensions/vscode/src/extension.ts]: thsi debug stuff does not work. => remove the contents of the []

File: extensions/vscode/src/extension.ts [hunk 3]
```diff
@@ -1,41 +1,49 @@
...
+
+import * as path from 'path';
+import * as os from 'os';
+
+import {Trace} from 'vscode-jsonrpc';
+import { commands, window, workspace, ExtensionContext, Uri } from 'vscode';
+import { LanguageClient, LanguageClientOptions, ServerOptions } from 'vscode-languageclient';
+
+export function activate(context: ExtensionContext) {
+    // The server is a locally installed in src/mydsl
```
Comment (cdietrich) [inline extensions/vscode/src/extension.ts]: should read src/ls

**Comment by andreasmuelder** [comment]
Thanks @cdietrich !

**Pull request merged**